django/db/migrations/operations/models.py

```diff
@@ -307,6 +307,25 @@ def state_forwards(self, app_label, state):
                 new_fields.append((name, field))
             state.models[related_key].fields = new_fields
             state.reload_model(*related_key)
+        # Repoint M2Ms with through pointing to us
+        related_models = {f.remote_field.model for f in model._meta.fields if getattr(f.remote_field, 'model', None)}
+        model_name = "%s.%s" % (app_label, self.old_name)
+        for related_model in related_models:
+            if related_model == model:
+                related_key = (app_label, self.new_name_lower)
```

Comment:
I could add a test, but I'm not sure what I would be testing for.
For this line to really matter, a model should have a M2M with through pointing at itself. But I can't imagine this scenario being useful.
A `continue` instead of this line would work for all other cases.
If I remove this branch and do the else branch unconditionally, another test fails (test_rename_model_with_self_referential_fk).